internal/engine/image_build_and_deployer.go

```diff
@@ -346,10 +355,12 @@ func (ibd *ImageBuildAndDeployer) createEntitiesToDeploy(ctx context.Context,
 					var sidecarInjected bool
 					e, sidecarInjected, err = sidecar.InjectSyncletSidecar(e, injectedRefSelector, ibd.syncletContainer)
 					if err != nil {
-						return nil, err
+						return result, err
 					}
 					if !sidecarInjected {
-						return nil, fmt.Errorf("Could not inject synclet: %v", e)
+						result.syncletInjectionError = true
+					} else {
+						result.syncletInjected = true
 					}
 					injectedSynclet = true
```

Comment:
seems like `sidecarInjected` and `injectedSynclet` variables are redundant?

Reply:
part of the confusion here is that there's a loop, and one variable is for the outer loop and one is for the inner loop. I broke this into a separate function, to avoid variables both inside and outside the loop